### DISCUSSION TITLE
Use React Components for Contextmenu

### DISCUSSION THREAD

**Discussion by zjda** [antvis/G6#5682] [Q&A]
Dear G6 Team and community,

Thank you for the great graph visualization engine!

As far as I can tell, the version 5.x has the same API for Contextmenu Plugin, where getContent expects a string or HTMLElement. It would be helpful if I could use React components (e.g. AntD menu) for the contextmenu to have a consistent appearance. I know that I can use graph.on("contextmenu") event to show any React components, but then I have to write extra code to manage the contextmenu (e.g. positioning to avoid clipping and mouse events to close the menu). I would appreciate it if the plugin could be improved or any workaround could be recommended.

Cheers,
-ZJ

**Comment by hustcc** [comment]
You can use `ReactDOM.render` to render a React Component before return a HTMLElement.
